### PR TITLE
add vast example to infra

### PR DESCRIPTION
**Pull request by aowen14**
Adding Vast as an example for infra

### CODE REVIEW

File: infra/vast/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+Deployment via (Vast.ai)[vast.ai]
+
+## Local CLI:
+Vast offers a nice UI to deploy instances, and you can use it with the command details below
+```bash
+vastai search offers 'compute_cap > 800 gpu_ram > 20 num_gpus = 1 static_ip=true direct_port_count > 1' 
+# Select the instance and paste it in for <instance-id> to launch
+# Single embedding model deployment:
+vastai create instance <instance-id> --image michaelf34/infinity:latest --env '-p 8000:8000' --disk 40 --args v2 --model-id michaelfeil/bge-small-en-v1.5 --port 8000
```
Comment (michaelfeil) [inline infra/vast/README.md:9]: Consider using `--api-key` in this example.

File: infra/vast/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+Deployment via (Vast.ai)[vast.ai]
+
+## Local CLI:
+Vast offers a nice UI to deploy instances, and you can use it with the command details below
```
Comment (michaelfeil) [inline infra/vast/README.md:4]: Maybe explain that you used the vastai python package?

File: infra/vast/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+## Finding your IP Address:
+Once your instance is up, the ports will be ready. In the Vast UI, you will see a blue button with an IP address and port range: ![vast_instance_view](vast_instance_view.png)
+
+In the view that pops up, you'll see something like: 
+
+```
+Open Ports
+XX.XX.XXX.XX:YYYY ->  8000/tcp
```
Comment (michaelfeil) [inline infra/vast/README.md:21]: Is there an option to forward not expose IPv4 and generate a dynamic adress. `http://some-random-name.ports.vast.ai`? @aowen14

File: infra/vast/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+Deployment via (Vast.ai)[vast.ai]
```
[the remark this answers is not in the document]
Comment (michaelfeil) [reply infra/vast/README.md:1]: @aowen14

**Review by michaelfeil — APPROVED** [review]
Approved, please resolve the last comments before merging.

File: infra/vast/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (aowen14) [reply infra/vast/README.md:1]: Sorry, didn't fix this before merge. Submitted another small PR for this!

**Pull request merged**